### PR TITLE
Fix issue 2269 - And a lot of other things

### PR DESCRIPTION
**Pull request by OsaAjani**
TL;DR: Transparency was broken all around, I fixed it.

What started as a "quick fix" for issue #2269 rapidly escalated into (re)discovery (see https://github.com/Zulko/moviepy/issues/2012#issuecomment-1642997120) of multiple bugs in regards to transparency handling in moviepy. To summarize, transparency was plain and simple broken :
- Reading video with transparency didn't work
- Writing video with background transparency didn't work
- Calculating final transparency of composite clip when the final clip should be partly transparent didn't work
- Using composite clip of composite clips whom add some clips with mask didn't work (original #2269)
- Compositing and bliting was partially buggy

Basically the only part that was working as intended was composite clip with basic clips inside, and a fully opaque background.

I also added codec extraction to FFMPEGParser when possible, as this info was required for proper transparency handling on ffmpeg.

I think this should fix not only #2269, but also #2109, #2082, #2008, #1978, #1935, #1881, #1821, #1759, #1301, #896, #400, and probably others I just dont know about...

- [x] I have provided code that clearly demonstrates the bug and that only works correctly when applying this fix
- [x] I have added suitable tests demonstrating a fixed bug or new/changed feature to the test suite in `tests/`
- [x] I have properly documented new or changed features in the documentation or in the docstrings
- [x] I have properly explained unusual or unexpected code in the comments around it

### CODE REVIEW

**Comment by OsaAjani** [comment]
I will give it a few days for other to look at the PR and flag anything they think is wrong, then if not protest will proceed with merging !

**Comment by OsaAjani** [comment]
Hey @Zulko @keikoro and others, if someone want to take a look at this PR to check I didn't do anything stupid I intend to merge it during the next two days without responses :)

**Comment by claell** [comment]
@OsaAjani will this also fix #195 (and incorporate some solution as was apparently proposed in #1395)?

File: moviepy/video/compositing/CompositeVideoClip.py [hunk 1]
```diff
@@ -67,6 +67,8 @@ def __init__(
         if bg_color is None:
             bg_color = 0.0 if is_mask else (0, 0, 0)
 
+        print(clips)
```
Comment (Implosiv3) [inline moviepy/video/compositing/CompositeVideoClip.py]: Print here.

File: moviepy/video/compositing/CompositeVideoClip.py
Comment (Implosiv3) [inline moviepy/video/compositing/CompositeVideoClip.py:1]: In new 302 line, there is a bug. It should be `mask = clip.mask or ColorClip(clip.size, color=1, is_mask=True)` instead of the `[1, 1]` mask size.

You can see the issue here (https://github.com/Zulko/moviepy/issues/2247) and my PR here (https://github.com/Zulko/moviepy/pull/2262).

Comment (OsaAjani) [reply moviepy/video/compositing/CompositeVideoClip.py:1]: Thanks, I will fix this and include the changes in your PR in this one, as well as adding a unit test

**Comment by OsaAjani** [comment]
I'll publish with a new tag 2.1.2 tomorrow

**Comment by claell** [comment]
@OsaAjani It seems that the release takes a little longer. Can you share a new (rough) ETA?

**Comment by OsaAjani** [comment]
Sorry, last days where more about home fixing than coding ^^. Definitely in the week.

**Comment by claell** [comment]
No problem, thanks for the update! Was just checking frequently the past days to see whether it already got released. 😄

**Comment by OsaAjani** [comment]
Done.

**Comment by claell** [comment]
Thanks! Looks as if the publish to PyPI failed, though? The logs indicate that the file already existed, or something like that.

**Comment by OsaAjani** [comment]
I fixed it

**Comment by claell** [comment]
Awesome! Just updated with pip and can confirm that the transparency problems in #2269 are fixed now, for me!
